SageMaker Endpoint magic command support (#215)
jupyterlab/jupyter-ai: docs/source/users/index.md

    alt='Screen shot of an example "Hello world" message sent to Jupyternaut, who responds with "Hello world, how are you today?"'
    class="screenshot" />

### Using the chat interrface with SageMaker endpoints

Jupyter AI supports language models hosted on SageMaker endpoints that use JSON

> APIs. The first step is to authenticate with AWS via the `boto3` SDK and have

Reviewer: ```suggestion
schemas. The first step is to authenticate with AWS via the `boto3` SDK and have
```

Aliases' names can contain ASCII letters (uppercase and lowercase), numbers, hyphens, underscores, and periods. They may not contain colons. They may also not override built-in commands — run `%ai help` for a list of these commands.

Aliases must refer to models or `LLMChain` objects; they cannot refer to other aliases.

### Using the chat interface with SageMaker endpoints

You can use the chat interface with models hosted using Amazon SageMaker.

> First, make sure that you've set your `AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY`, and `AWS_SESSION_TOKEN` environment variables either before starting JupyterLab or using the `%env` magic command within JupyterLab.

Reviewer: `AWS_SESSION_TOKEN` might be internal only. Usually, `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` are sufficient for most users. Rather than going into specific details which might differ between users, we should link to the AWS CLI documentation page, where all these options are listed and add a note that the access to AWS should be configured before proceeding with using the magics statement.
https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-configure.html

> Jupyter AI supports language models hosted on SageMaker endpoints that use JSON APIs. Authenticate with AWS via the `boto3` SDK and have the credentials stored in the `default` profile.  Guidance on how to do this can be found in the [`boto3` documentation](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/credentials.html).

Reviewer: I think this para is sufficient to providing info on setting credentials.
```suggestion
Jupyter AI supports language models hosted on SageMaker endpoints that use JSON schemas. Authenticate with AWS via the `boto3` SDK and have the credentials stored in the `default` profile.  Guidance on how to do this can be found in the [`boto3` documentation](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/credentials.html).
```

You will need to deploy a model in SageMaker, then provide it as your model name (as `sagemaker-endpoint:my-model-name`). See the [documentation on how to deploy a JumpStart model](https://docs.aws.amazon.com/sagemaker/latest/dg/jumpstart-deploy.html).

All SageMaker endpoint requests require you to specify the `--region-name`, `--request-schema`, and `--response-path` options. The example below presumes that you have deployed a model called `jumpstart-dft-hf-text2text-flan-t5-xl`.

```
%%ai sagemaker-endpoint:jumpstart-dft-hf-text2text-flan-t5-xl --region-name=us-east-1 --request-schema={"text_inputs":"<prompt>"} --response-path=generated_texts.[0] -f code
Write Python code to print "Hello world"
```

The `--region-name` parameter is set to the [AWS region code](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-regions-availability-zones.html) where the model is deployed, which in this case is `us-east-1`.

> The `--request-schema` parameter is the JSON object the endpoint expects, with the prompt being substituted into any value that matches the string literal `"<prompt>"`. For example, the request schema `{"text_inputs":"<prompt>"}` generates a JSON object with the prompt stored under the `text_inputs` key.

Reviewer: ```suggestion
The `--request-schema` parameter is the JSON object the endpoint expects as input, with the prompt being substituted into any value that matches the string literal `"<prompt>"`. For example, the request schema `{"text_inputs":"<prompt>"}` will submit a JSON object with the prompt stored under the `text_inputs` key.
```
